src/devices/Ft4222/FtFunction.cs

```diff
@@ -12,15 +12,133 @@ namespace Iot.Device.Ft4222
     /// </summary>
     internal class FtFunction
     {
-        #region ftd2xx.dll
+        #region ftd2xx wrapper
 
         /// <summary>
         /// Create Device Information List
         /// </summary>
         /// <param name="numdevs">number of devices</param>
         /// <returns>The status</returns>
-        [DllImport("ftd2xx")]
-        public static extern FtStatus FT_CreateDeviceInfoList(out uint numdevs);
+        public static FtStatus FT_CreateDeviceInfoList(out uint numdevs)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                return FT_CreateDeviceInfoListLinux(out numdevs);
+            }
+
+            return FT_CreateDeviceInfoListWin(out numdevs);
+        }
+
+        /// <summary>
+        /// Get Device Information Detail
+        /// </summary>
+        /// <param name="index">Index of the device</param>
+        /// <param name="flags">Flags</param>
+        /// <param name="chiptype">Device type</param>
+        /// <param name="id">ID</param>
+        /// <param name="locid">Location ID</param>
+        /// <param name="serialnumber">Serial Number</param>
+        /// <param name="description">Description</param>
+        /// <param name="ftHandle">Handle</param>
+        /// <returns>The status</returns>
+        public static FtStatus FT_GetDeviceInfoDetail(uint index, out uint flags, out FtDevice chiptype, out uint id, out uint locid, in byte serialnumber, in byte description, out IntPtr ftHandle)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
```

Comment:
Ideally we would make this check only once and cache the result in a static or something.

Reply:
do you mean the delegate? I think enum comparison is more or less as fast as null check. Other option is to use interop interface and implement it per platform

Reply:
Those functions are called like 2 or 3 times maximum in all the life cycle of the code. So I won't try to over complicated everything. And they are called during the initialization phase.

Reply:
I'm fine with keeping this as is.